Makefile

```diff
@@ -136,12 +136,12 @@ $(HCLOUD):
 KIND := $(abspath $(TOOLS_BIN_DIR)/kind)
 kind: $(KIND) ## Build a local copy of kind
 $(KIND):
-	go install sigs.k8s.io/kind@v0.20.0
+	go install sigs.k8s.io/kind@v0.22.0
 
 KUBECTL := $(abspath $(TOOLS_BIN_DIR)/kubectl)
 kubectl: $(KUBECTL) ## Build a local copy of kubectl
 $(KUBECTL):
-	curl -fsSL "https://dl.k8s.io/release/v1.28.4/bin/$$(go env GOOS)/$$(go env GOARCH)/kubectl" -o $(KUBECTL)
+	curl -fsSL "https://dl.k8s.io/release/v1.29.4/bin/$$(go env GOOS)/$$(go env GOARCH)/kubectl" -o $(KUBECTL)
```

Comment:
shouldn't this be 1.29.3? you have changes from 28.4 to 29.4 multiple times. If I'm not mistaken it should almost always be 29.3 instead

Reply:
From [Cluster API docs](https://main.cluster-api.sigs.k8s.io/reference/versions): 

For example, Cluster API v1.7.0 would support the following Kubernetes versions:

    v1.26.x to v1.29.x for the management cluster
    v1.24.x to v1.29.x for the workload cluster
    When Kubernetes 1.30 is released, it will be supported in v1.7.x (but not in v1.6.x)


I think it's okay go proceed with 1.29.4 despite consuming 1.29.3 go dependencies here. 1.7.x version of cluster API supports 1.29.x version of kubernetes.

Reply:
and why do you not want to go to 1.29.4 go dependencies?

Reply:
Cluster API is using 0.29.3 as dependencies on release-1.7 branch and I want to have that in sync. 

https://github.com/kubernetes-sigs/cluster-api/blob/c4fc57b89756cc1a602a29ae7d6e583523a1933e/go.mod#L38-L47

I want only go dependencies to be in sync with cluster API release-1.7 branch and want to use 1.29.4 which is latest patch available in Kubernetes 1.29 series for kubectl and creating management cluster.

Reply:
well I cannot argue against that if this is important to you